src/Controls/src/Core/Element.cs

```diff
@@ -118,13 +211,10 @@ internal IEnumerable<Element> AllChildren
 		// return null by default so we don't need to foreach over an empty collection in OnPropertyChanged
 		internal virtual IEnumerable<Element> ChildrenNotDrawnByThisElement => null;
 
-		IReadOnlyList<Element> IElementController.LogicalChildren => LogicalChildrenInternal;
+		IReadOnlyList<Element> _logicalChildrenInternal;
+		IList<Element> _internalChildren;
 
-		/// <include file="../../docs/Microsoft.Maui.Controls/Element.xml" path="//Member[@MemberName='LogicalChildren']/Docs/*" />
-		[EditorBrowsable(EditorBrowsableState.Never)]
-		[Obsolete("Do not use! This is to be removed! Just used by Hot Reload! To be replaced with IVisualTreeElement!")]
-		public ReadOnlyCollection<Element> LogicalChildren =>
-			new ReadOnlyCollection<Element>(new TemporaryWrapper(LogicalChildrenInternal));
+		IReadOnlyList<Element> IElementController.LogicalChildren => LogicalChildrenInternal;
```

Comment:
Wow, gone from DO NOT USE to a public feature. Is this intentional and why was it marked so strongly before?

Reply:
Yea, I don't fully understand the strong language. IVT really just exists because of where `LogicalChildren` was setup wrong. Any case where the `LogicalChildren` and the IVT diverge seems like a defect of the control. For example, `ListView`, for some reason, completely breaks if you start mapping the `LogicalChildren` to realized cells (basically what we do for CV). 

I think we could get rid of all these hacks https://github.com/dotnet/maui/blob/ebfa27a39951e526a04f5edbff699b7a3fd44b44/src/Controls/src/Core/BindableObject.cs#L265-L279 if the `LogicalChildren` were accurate.

Is there any scenario where it makes sense for `LogicalChildren` != IVT.GetChildren?

I could rename `LogicalChildren` to `VisualChildren` and then I guess it's more in line with the language we're using for IVT.

Reply:
AFAIK, the original reason why LogicalChildren was discouraged was to avoid people using it to access underlying elements and build UI Structures around it rather than using Binding and acting on elements organically. For example, you wouldn't want someone to try to poke at individual elements in a ListView or CollectionView via LogicalChildren, as those items come and go as needed, and trying to pin something to one could create problems if you don't know what you're doing.

IMO "VisualChildren" is a misnomer. I believe @PureWeen brought that up before with Shell: Shell is virtual controls made up of visual elements that are not, by themselves, "Visual." But they still need to be represented in the LVT as that is the proper makeup of your application, From what the Window has on down. Then again, IMO, as long as the information is correct, I don't care what the name is, lol.